scripts/screeps_rl_training_runner.py

```
@@ -3349,6 +3382,196 @@ def policy_update_scalar_reward(return_tuple: Sequence[Any], weights: JsonObject
+
+
+def policy_update_gradient_momentum_from_state_payload(raw: Any) -> JsonObject | None:
+    if not isinstance(raw, dict):
+        return None
+    if text_or_none(raw.get("type")) == GRADIENT_MOMENTUM_EVIDENCE_TYPE:
+        return copy.deepcopy(raw)
+    momentum = first_mapping(raw, ("gradientMomentum", "gradient_momentum", "momentum"))
+    return copy.deepcopy(momentum) if momentum is not None else None
+
+
+def load_policy_gradient_momentum_state(policy_gradient: JsonObject | None, out_dir: Path) -> JsonObject | None:
```

Ruff (0.15.15)

[warning] Too many return statements (9 > 6)

(PLR0911)